core/chains/evm/logpoller/log_poller.go

```diff
@@ -93,7 +94,7 @@ type ReplayRequest struct {
 // - 1 db tx including block write and logs write to logs.
 // How fast that can be done depends largely on network speed and DB, but even for the fastest
 // support chain, polygon, which has 2s block times, we need RPCs roughly with <= 500ms latency
-func NewLogPoller(orm *ORM, ec client.Client, lggr logger.Logger, pollPeriod time.Duration, finalityDepth, backfillBatchSize, rpcBatchSize int64) *logPoller {
+func NewLogPoller(orm *ORM, ec client.Client, lggr logger.Logger, pollPeriod time.Duration, finalityDepth int64, backfillBatchSize int64, rpcBatchSize int64, keepBlocksDepth int64) *logPoller {
```

Comment:
Minor note/nit #1: This constructor signature is quite long, worth doing something to shorten it? We don't really have a paradigm for this in core, but just something I noticed in other places as well. In other large Go codebases I've worked on, we actually didn't have these `New*` methods at all and almost all fields were public (but this was also the case because we were using https://github.com/uber-go/fx).

 Minor note/nit #2: this `New`er returns the private `*logPoller`, but wouldn't it be more idiomatic if it returned the public interface type instead? (`LogPoller`)

Reply:
1 - I like explicit args as its just harder to misuse, you can't forget to specify a struct field this way. Lots of internal fields that I wouldn't want to expose (cached values etc.). Std go library has NewXX constructors all over it.
2 - Its more flexible to return the struct, callers can implicitly cast as needed e.g. ```var lp LogPoller = NewLogPoller```. General go idiom is "accept interfaces, return structs"